core-bundle/src/Twig/Loader/TemplateLocator.php

```diff
@@ -114,6 +113,8 @@ public function findTemplates(string $path): array
             $templates[Path::normalize($file->getRelativePathname())] = Path::canonicalize($file->getPathname());
         }
 
+        ksort($templates);
```

Comment:
Let's add a comment, so that we understand why we are doing it this way in the future.

```suggestion
        // For some reason Finder#sortByName() is very slow on Windows, so we
        // manually sort the results instead.
        ksort($templates);
```

Reply:
Are there no improvements on Unix at all? (May be you would also need to test with lots of extensions, i.e. lots of template directories.)

Reply:
Indeed, I also get a ~30% speedup over using `Finder#sortByName()`. I adjusted the above suggestion accordingly.

Reply:
What does the `Finder` class do differently?

Reply:
It uses `uasort` with a static callback that looks like [this](https://github.com/symfony/symfony/blob/265b5d0edb2c0dd8e33ad71d99c6aa8b2c016927/src/Symfony/Component/Finder/Iterator/SortableIterator.php#L51):

```php
static fn (\SplFileInfo $a, \SplFileInfo $b) => $order * strcmp($a->getRealPath() ?: $a->getPathname(), $b->getRealPath() ?: $b->getPathname())
```

Reply:
And doesn't this mean that `ksort()` will probably not work with symlinks?

Reply:
Why would it not work with symlinks?

Reply:
I figured that is why Symfony is using `getRealPath()`?

Reply:
For our purposes we only need to sort by the relative path. If you had a template folder that actually symlinks to somewhere outside the project root for some reason, previously the result would be "wrong" as `->sortByName()` uses the real path.

Reply:
Well then…